pkg/kubectl/cmd/util/editor/editoptions.go

```diff
@@ -478,14 +577,17 @@ func (o *EditOptions) visitToCreate(createVisitor resource.Visitor) error {
 func (o *EditOptions) visitAnnotation(annotationVisitor resource.Visitor) error {
 	// iterate through all items to apply annotations
 	err := annotationVisitor.Visit(func(info *resource.Info, incomingErr error) error {
-		// put configuration annotation in "updates"
-		if o.ApplyAnnotation {
-			if err := kubectl.CreateOrUpdateAnnotation(true, info, o.Encoder); err != nil {
+		if o.Record || cmdutil.ContainsChangeCause(info) {
+			if err := cmdutil.RecordChangeCause(info.Object, o.ChangeCause); err != nil {
 				return err
 			}
 		}
-		if o.Record || cmdutil.ContainsChangeCause(info) {
-			if err := cmdutil.RecordChangeCause(info.Object, o.ChangeCause); err != nil {
+		if o.EditMode == ApplyEditMode {
```

Comment:
so your suggestion is like branching ?
```console
if o.EditMode != ApplyEditMode {
	// put configuration annotation in "updates"
	if o.ApplyAnnotation {
		if err := kubectl.CreateOrUpdateAnnotation(true, info, o.Encoder); err != nil {
			return err
		}
	}
}

if o.Record || cmdutil.ContainsChangeCause(info) {
	if err := cmdutil.RecordChangeCause(info.Object, o.ChangeCause); err != nil {
		return err
	}
}
```
or could we keep the original one, since I think `o.ApplyAnnotation` will never be set if we using apply-edit ?
```console
if o.ApplyAnnotation {            ------> //always pass here if we using apply edit
	if err := kubectl.CreateOrUpdateAnnotation(true, info, o.Encoder); err != nil {
		return err
	}
}
if o.Record || cmdutil.ContainsChangeCause(info) {
	if err := cmdutil.RecordChangeCause(info.Object, o.ChangeCause); err != nil {
		return err
	}
}
```